src/test/java/com/gmail/woosay333/onlinebookstore/repository/book/BookRepositoryTest.java

```diff
@@ -0,0 +1,184 @@
+
+        assertTrue(actual.isEmpty(), "Optional of book by non-existing ISBN should be empty");
+    }
+
+    @Test
+    @DisplayName("""
+            Find book with categories by existing ID,
+            return optional of book with set of categories
+            """)
+    void findByIdWithCategories_FindingBookByExistingId_ReturnOptionalOfBook() {
+        Optional<Book> actual = bookRepository.findByIdWithCategories(VALID_BOOK_ID_HOBBIT);
+
+        assertTrue(actual.isPresent(), "Optional of book by existing ID should be present");
+        assertNotNull(actual.get().getCategories());
+    }
+
+    @Test
+    @DisplayName("""
+            Find book with categories by non existing ID, return empty optional
+            """)
+    void findByIdWithCategories_FindingBookByNonExistingId_ReturnEmptyOptional() {
+        Optional<Book> actual = bookRepository.findByIdWithCategories(INVALID_BOOK_ID_10L);
+
+        assertTrue(actual.isEmpty(), "Optional of book by non-existing ID should be empty");
+    }
+
+    @Test
+    @DisplayName("""
+            Find all books with categories by params, return page of books
+            """)
+    void findAllByParams_ReturnAllBooksByParams() {
+        Specification<Book> specification = Specification.where(null);
+        Pageable pageable = Pageable.unpaged();
+
+        Page<Book> actual = bookRepository.findAll(specification, pageable);
+        int expectedSize = 3;
+
+        assertEquals(expectedSize, actual.getSize());
+        actual.stream()
+                .map(Book::getCategories)
+                .forEach(Assertions::assertNotNull);
+    }
+
+    @Test
+    @DisplayName("""
+            Find all books with categories, returns list of books
+            """)
+    void findAllBooks_GetAllBooks_ReturnAllBooks() {
+        Pageable pageable = Pageable.unpaged();
+
+        List<Book> actual = bookRepository.findAllBooks(pageable);
+        int expectedSize = 3;
+
+        assertEquals(expectedSize, actual.size());
+        actual.stream()
+                .map(Book::getCategories)
+                .forEach(Assertions::assertNotNull);
+    }
+
+    @Test
+    @DisplayName("""
+            Find all books witch belongs to category ID, returns list of books
+            """)
+    void findAllByCategoryId_ExistCategoryId_ReturnAllBooksByCategoryId() {
+        Pageable pageable = Pageable.unpaged();
+        Long fantasyId = 1L;
+        Long novelId = 2L;
+
+        List<Book> actualFantasy = bookRepository.findAllByCategories_Id(fantasyId, pageable);
+        List<Book> actualNovel = bookRepository.findAllByCategories_Id(novelId, pageable);
+
+        assertNotNull(actualFantasy);
+        assertNotNull(actualNovel);
+
+        int expectedSizeFantasy = 2;
+        int expectedSizeNovel = 1;
+
+        assertEquals(expectedSizeFantasy, actualFantasy.size(), "Size should be equals");
+        assertEquals(expectedSizeNovel, actualNovel.size(), "Size should be equals");
```

Comment:
same here and in other methods
Hint: you can use `assertThat(actualFantazy).hasSize(2)` and you will have a proper message by default